paimon-core/src/main/java/org/apache/paimon/schema/SchemaValidation.java

```diff
@@ -675,4 +679,26 @@ private static void validateIncrementalClustering(TableSchema schema, CoreOption
                     PRIMARY_KEY.key());
         }
     }
+
+    public static void validateChainTableOptions(
+            Map<String, String> tableOptions, @Nullable String primaryKeys, boolean partitionTbl) {
+        if (Boolean.parseBoolean(tableOptions.get(CoreOptions.CHAIN_TABLE_ENABLED.key()))) {
+            Options options = Options.fromMap(tableOptions);
```

Comment:
I think we should also check that the `changelog-producer != ("lookup", "full-compaction")`